osquery/core/windows/process.cpp

```diff
@@ -0,0 +1,209 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <sstream>
+#include <vector>
+
+#include <signal.h>
+
+#include <sys/types.h>
+#include <boost/algorithm/string.hpp>
+
+#include "osquery/core/process.h"
+
+namespace {
+osquery::PlatformPidType __declspec(nothrow) duplicateHandle(osquery::PlatformPidType src) {
+  osquery::PlatformPidType handle = osquery::kInvalidPid;
+
+  if (src != osquery::kInvalidPid) {
+    if (!::DuplicateHandle(GetCurrentProcess(),
+                           src,
+                           GetCurrentProcess(),
+                           &handle,
+                           0,
+                           FALSE,
+                           DUPLICATE_SAME_ACCESS)) {
+      handle = osquery::kInvalidPid;
+    }
+  }
+  return handle;
+}
+}
```

Comment:
The intention here was to create a `duplicateHandle` that would be accessible only within the scope of the file. Would you prefer a static function instead outside the `osquery` namespace?